numba/tests/test_flags.py

```diff
@@ -0,0 +1,39 @@
+import unittest
+
+import numba
+from numba import jit
+from numba.core.targetconfig import Option
+
+from numba.core.compiler import Flags
+from numba.typed.typedlist import List
+
+
+class MyFlags(Flags):
+    extra_option = Option(
+        type=bool,
+        default=False,
+        doc="define extra option for further extension",
+    )
+
+
+class MyTestCase(unittest.TestCase):
+
+    def test_define_flag_class(self):
+        my_flags = MyFlags()
+
+        # extremely strange. If I put empty_list outside JIT, segfaults.
+        # l = List.empty_list(numba.int64)
+
+        @jit(flags=my_flags, debug=True)
+        def foo():
+            # put it here, don't segfault.
+            l = List.empty_list(numba.int64)
```

Comment:
this segfault problem is solved by: https://github.com/numba/numba/pull/8000/files#r859196496

Reply:
Many thanks for your time! I updated these codes as u pointed out.